lib/Conversion/TorchToLinalg/DataMovement.cpp

```diff
@@ -748,39 +807,14 @@ class ConvertAtenSliceTensorOp : public OpConversionPattern<AtenSliceTensorOp> {
     SmallVector<Value> inputShape = getTensorSizes(rewriter, loc, input);
     Value dimSize = inputShape[dim];
 
-    auto adjustStartOrEnd = [&](Value startOrEndTorchType,
-                                Value startOrEndBuiltin, Value valueForNone) {
-      if (startOrEndTorchType.getType().isa<Torch::NoneType>())
-        return valueForNone;
-      auto dimSizeAsInt = castIndexToInt64(rewriter, loc, dimSize);
-      Value startOrEndToPositive =
-          toPositiveDimDynamic(rewriter, loc, startOrEndBuiltin, dimSizeAsInt);
-      // startOrEnd < 0 ? 0 : startOrEnd
-      Value cst0 = rewriter.create<arith::ConstantOp>(
-          loc, rewriter.getZeroAttr(dimSizeAsInt.getType()));
-      Value predDimSltZero = rewriter.create<arith::CmpIOp>(
-          loc, arith::CmpIPredicate::slt, startOrEndToPositive, cst0);
-      Value startOrEndAtLeastZero = rewriter.create<arith::SelectOp>(
-          loc, predDimSltZero, cst0, startOrEndToPositive);
-      // startOrEnd > dimSizeAsInt ? dimSizeAsInt : startOrEnd
-      Value startOrEndSgtDimSize = rewriter.create<arith::CmpIOp>(
-          loc, arith::CmpIPredicate::sgt, startOrEndAtLeastZero, dimSizeAsInt);
-      Value startOrEndBoundedByDimSize = rewriter.create<arith::SelectOp>(
-          loc, startOrEndSgtDimSize, dimSizeAsInt, startOrEndAtLeastZero);
-
-      return castIntToIndex(rewriter, loc, startOrEndBoundedByDimSize);
-    };
+    Value startTorchType = op.start();
+    Value endTorchType = op.end();
+    Value startBuiltinType = adaptor.start();
+    Value endBuiltinType = adaptor.end();
```

Comment:
Same comment as in `prepareArgumentsForSlicingOp`, I would place the type info first, and end with `start` or `end`